File: backend/java-springboot-hexagonal/src/main/java/com/sakrafux/realworld/application/service/TagService.java
package com.sakrafux.realworld.application.service;

import com.sakrafux.realworld.application.port.in.GetTagsQuery;
import com.sakrafux.realworld.application.port.out.TagRepository;
import com.sakrafux.realworld.domain.model.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class TagService implements GetTagsQuery {

    private final TagRepository tagRepository;

    @Override
    @Transactional(readOnly = true)
    public List<Tag> getTags() {
        return tagRepository.findAll();
    }
}

File: backend/go-chi-hexagonal/internal/application/service/tag_service.go
package service

import (
	"context"

	"github.com/Sakrafux/realworld-tech-comparison/backend/go-chi-hexagonal/internal/application/port"
	"github.com/Sakrafux/realworld-tech-comparison/backend/go-chi-hexagonal/internal/domain"
)

type tagService struct {
	tagRepo port.TagRepository
}

func NewTagService(tagRepo port.TagRepository) port.TagService {
	return &tagService{
		tagRepo: tagRepo,
	}
}

// GetTags retrieves all tags by delegating to the repository.
func (s *tagService) GetTags(ctx context.Context) ([]domain.Tag, error) {
	return s.tagRepo.FindAll(ctx)
}
